libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -182,7 +194,17 @@ static void __pthread_exit(void *result)
 	}
 #else
 	if (state==DT_DETACHED && self->map_base) {
-		free(self->map_base);
+		// As we use malloc/free which is considerably more complex
+		// than mmap/munmap to call and can even require a valid
+		// thread context, it's difficult to implement __unmapself.
+		//
+		// Here we take an alternative approach which simply defers
+		// the deallocation. An obvious downside of this approach is
+		// that it keeps the stack longer. (possibly forever.)
+		// To avoid wasting too much memory, we keep the free queue
+		// length at most one.
```

Comment:
Can we be consistent about C vs C++ comments?  Below you've used C multi-line comment.

Reply:
i followed nearby code.

Reply:
fixed